spec/navigation.spec.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react"
+import { NostoCategory, NostoHome, NostoPlacement, NostoProduct, NostoProvider } from "../src"
+import RecommendationComponent from "./renderer"
+import { Link, BrowserRouter, Route, Routes, useParams } from "react-router-dom"
+import { fireEvent, render, screen, waitFor } from "@testing-library/react"
+import { WAIT_FOR_TIMEOUT } from "./utils"
+import { categoryEvent, frontEvent, productEvent } from "./events"
+
+function HomePage() {
+  return <>
+    <NostoPlacement id="frontpage-nosto-1" />
+    <NostoPlacement id="frontpage-nosto-3" />
+    <NostoPlacement id="frontpage-nosto-4" />
+    <NostoHome />
+    <Link to="/collections/hoodies">Hoodies</Link>
+  </>    
+}
+
+function CategoryPage() {
+  const { category } = useParams()
+  return <>
+    <NostoPlacement id="categorypage-nosto-1" />
+    <NostoPlacement id="categorypage-nosto-2" />
+    <NostoCategory category={category!} />
+    <Link to="/products/123">Product 123</Link>
+    <Link to="/products/234">Product 234</Link>
+    <Link to="/">Home</Link>
+  </>
+}
+
+function ProductPage() {
+  const { product } = useParams()
+  return <>
+    <NostoPlacement id="productpage-nosto-1" />
+    <NostoPlacement id="productpage-nosto-2" />
+    <NostoPlacement id="productpage-nosto-3" />
+    <NostoProduct product={product!} />
+    <Link to="/products/234">Product 234</Link>
+    <Link to="/collections/hoodies">Hoodies</Link>
+    <Link to="/">Home</Link>
+  </>
+}
+
+function Main() {
+  return <NostoProvider
+    account="shopify-11368366139"
+    recommendationComponent={<RecommendationComponent />}> 
+    <BrowserRouter>
```

Comment:
We need to keep in mind this only checks for navigation using the browser's history.

Reply:
can you elaborate?